In [197]:
import pandas as pd  #Importing necessary packages

In [199]:
#importing .csv file to work on
survey_responses = pd.read_csv('/Users/anthonydaniels/Desktop/Udacity Data Science Study/Udacity Project 1/Udacity Project 1 csv dataset 1.csv')

In [201]:
survey_responses.head()  #look at first 5 lines of code



ResponseId                      MainBranch                 Age  \
0           1  I am a developer by profession  Under 18 years old   
1           2  I am a developer by profession     35-44 years old   
2           3  I am a developer by profession     45-54 years old   
3           4           I am learning to code     18-24 years old   
4           5  I am a developer by profession     18-24 years old   

            Employment RemoteWork   Check  \
0  Employed, full-time     Remote  Apples   
1  Employed, full-time     Remote  Apples   
2  Employed, full-time     Remote  Apples   
3   Student, full-time        NaN  Apples   
4   Student, full-time        NaN  Apples   

                                    CodingActivities  \
0                                              Hobby   
1  Hobby;Contribute to open-source projects;Other...   
2  Hobby;Contribute to open-source projects;Other...   
3                                                NaN   
4                                                NaN   

                                             EdLevel  \
0                          Primary/elementary school   
1       Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
2    Master’s degree (M.A., M.S., M.Eng., MBA, etc.)   
3  Some college/university study without earning ...   
4  Secondary school (e.g. American high school, G...   

                                           LearnCode  \
0                             Books / Physical media   
1  Books / Physical media;Colleague;On the job tr...   
2  Books / Physical media;Colleague;On the job tr...   
3  Other online resources (e.g., videos, blogs, f...   
4  Other online resources (e.g., videos, blogs, f...   

                                     LearnCodeOnline  ... JobSatPoints_6  \
0                                                NaN  ...            NaN   
1  Technical documentation;Blogs;Books;Written Tu...  ...            0.0   
2  Technical documentation;Blogs;Books;Written Tu...  ...            NaN   
3  Stack Overflow;How-to videos;Interactive tutorial  ...            NaN   
4  Technical documentation;Blogs;Written Tutorial...  ...            NaN   

  JobSatPoints_7 JobSatPoints_8 JobSatPoints_9 JobSatPoints_10  \
0            NaN            NaN            NaN             NaN   
1            0.0            0.0            0.0             0.0   
2            NaN            NaN            NaN             NaN   
3            NaN            NaN            NaN             NaN   
4            NaN            NaN            NaN             NaN   

  JobSatPoints_11           SurveyLength SurveyEase ConvertedCompYearly JobSat  
0             NaN                    NaN        NaN                 NaN    NaN  
1             0.0                    NaN        NaN                 NaN    NaN  
2             NaN  Appropriate in length       Easy                 NaN    NaN  
3             NaN               Too long       Easy                 NaN    NaN  
4             NaN              Too short       Easy                 NaN    NaN  

[5 rows x 114 columns]

In [57]:
survey_responses.describe() #descriptive statistics of file

ResponseId      CompTotal       WorkExp  JobSatPoints_1  \
count  65437.000000   3.374000e+04  29658.000000    29324.000000   
mean   32719.000000  2.963841e+145     11.466957       18.581094   
std    18890.179119  5.444117e+147      9.168709       25.966221   
min        1.000000   0.000000e+00      0.000000        0.000000   
25%    16360.000000   6.000000e+04      4.000000        0.000000   
50%    32719.000000   1.100000e+05      9.000000       10.000000   
75%    49078.000000   2.500000e+05     16.000000       22.000000   
max    65437.000000  1.000000e+150     50.000000      100.000000   

       JobSatPoints_4  JobSatPoints_5  JobSatPoints_6  JobSatPoints_7  \
count    29393.000000    29411.000000    29450.000000     29448.00000   
mean         7.522140       10.060857       24.343232        22.96522   
std         18.422661       21.833836       27.089360        27.01774   
min          0.000000        0.000000        0.000000         0.00000   
25%          0.000000        0.000000        0.000000         0.00000   
50%          0.000000        0.000000       20.000000        15.00000   
75%          5.000000       10.000000       30.000000        30.00000   
max        100.000000      100.000000      100.000000       100.00000   

       JobSatPoints_8  JobSatPoints_9  JobSatPoints_10  JobSatPoints_11  \
count    29456.000000    29456.000000     29450.000000     29445.000000   
mean        20.278165       16.169432        10.955713         9.953948   
std         26.108110       24.845032        22.906263        21.775652   
min          0.000000        0.000000         0.000000         0.000000   
25%          0.000000        0.000000         0.000000         0.000000   
50%         10.000000        5.000000         0.000000         0.000000   
75%         25.000000       20.000000        10.000000        10.000000   
max        100.000000      100.000000       100.000000       100.000000   

       ConvertedCompYearly        JobSat  
count         2.343500e+04  29126.000000  
mean          8.615529e+04      6.935041  
std           1.867570e+05      2.088259  
min           1.000000e+00      0.000000  
25%           3.271200e+04      6.000000  
50%           6.500000e+04      7.000000  
75%           1.079715e+05      8.000000  
max           1.625660e+07     10.000000

In [203]:
survey_responses.isna().sum() #check for nulls and count them

ResponseId                 0
MainBranch                 0
Age                        0
Employment                 0
RemoteWork             10631
                       ...  
JobSatPoints_11        35992
SurveyLength            9255
SurveyEase              9199
ConvertedCompYearly    42002
JobSat                 36311
Length: 114, dtype: int64

In [205]:
survey_responses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Columns: 114 entries, ResponseId to JobSat
dtypes: float64(13), int64(1), object(100)
memory usage: 56.9+ MB


We have 114 columns/features in our dataset.  We need to start the cleaning process.
Our project is going to attempt create a model that predicts those who will create a career which will be defined as those with over 10 years work experience.

In [207]:
# eliminate columns that we don't want to be taken into account for our model
# we have a column 'check' that appears to have responses all 'apples'.  check to see if there are any other responses

# Check for non-'apples' responses in the 'check' column
non_apples = survey_responses[survey_responses['Check'] != 'Apples']

# Display results
if len(non_apples) > 0:
    print(f"Found {len(non_apples)} rows with non-'apples' responses:")
    print(non_apples[['Check']])  # Show only the 'check' column
else:
    print("All responses in 'Check' column are 'Apples'.")

All responses in 'Check' column are 'Apples'.


Based on the fact that all of the responses are 'Apples' under the 'Check' column, we can assume that it is a nonimpactful column since it applies to everybody in the survey

In [209]:
#code to remove undesired columns from file
survey_responses.drop(columns=['Check', 'OrgSize', 'PurchaseInfluence', 'BuyNewTool', 'BuildvsBuy', 'TechEndorse', 'Currency','CompTotal', 'LanguageAdmired', 'DatabaseAdmired', 'PlatformAdmired', 'WebframeAdmired', 'EmbeddedAdmired', 'MiscTechWantToWorkWith', 'MiscTechAdmired', 'ToolsTechAdmired', 'NEWCollabToolsWantToWorkWith', 'NEWCollabToolsAdmired', 'OfficeStackAsyncWantToWorkWith', 'OfficeStackAsyncAdmired', 'OfficeStackSyncWantToWorkWith', 'OfficeStackSyncAdmired', 'AISearchDevWantToWorkWith', 'AISearchDevAdmired', 'SurveyLength', 'SurveyEase', 'ConvertedCompYearly'], inplace=True)

In [211]:
survey_responses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 87 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   ResponseId                      65437 non-null  int64  
 1   MainBranch                      65437 non-null  object 
 2   Age                             65437 non-null  object 
 3   Employment                      65437 non-null  object 
 4   RemoteWork                      54806 non-null  object 
 5   CodingActivities                54466 non-null  object 
 6   EdLevel                         60784 non-null  object 
 7   LearnCode                       60488 non-null  object 
 8   LearnCodeOnline                 49237 non-null  object 
 9   TechDoc                         40897 non-null  object 
 10  YearsCode                       59869 non-null  object 
 11  YearsCodePro                    51610 non-null  object 
 12  DevType                         

In [213]:
WorkExp_NA = survey_responses['WorkExp'].isna().sum()  #determine if there are NA to be dealth with in WorkExp column
print (WorkExp_NA)

35779


In [215]:
survey_responses['Career'] = survey_responses['WorkExp'].fillna(0) >=10   #creates new column named Career and assigns False to everything w/NA or Less than 10 years of work experience

In [217]:
print(survey_responses[['WorkExp','Career']].head()) ##checks WorkExp vs new Career column

   WorkExp  Career
0      NaN   False
1     17.0    True
2      NaN   False
3      NaN   False
4      NaN   False


In [219]:
survey_responses['WorkExp'] = survey_responses['WorkExp'].fillna(0)  #replaces all 'NaN' rows under 'WorkExp' to an assumed work experience of 0

In [221]:
survey_responses['MainBranch'] = survey_responses['MainBranch'].fillna('No_Experience') ### Replaces all 'Nan' responses in 'MainBranch' column to 'No_Experience'


In [225]:
###Converting Responses under 'MainBranch' column to individual columns with Boolean responses

Main_Branch_choices = [
    'I am a developer by profession',
    'I am not primarily a developer, but I write code sometimes as part of my work/studies',
    'I used to be a developer by profession, but no longer am',
    'I am learning to code',
    'I code primarily as a hobby',
    'No_Experience'
]

# Create boolean columns correctly
for option in Main_Branch_choices:
    # Improved column name cleaning
    col_name = (
        option.lower()  # Convert to lowercase
        .replace(' ', '_')
        .replace('/', '_')
        .replace(',', '')
        .replace("'", "")  # Remove apostrophes
    )
    # Compare against the current option only (not the whole list)
    survey_responses[f"Branch_{col_name}"] = (survey_responses['MainBranch'] == option)

In [227]:
survey_responses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 100 columns):
 #   Column                                                                                       Non-Null Count  Dtype  
---  ------                                                                                       --------------  -----  
 0   ResponseId                                                                                   65437 non-null  int64  
 1   MainBranch                                                                                   65437 non-null  object 
 2   Age                                                                                          65437 non-null  object 
 3   Employment                                                                                   65437 non-null  object 
 4   RemoteWork                                                                                   54806 non-null  object 
 5   CodingActivities               

In [107]:
survey_responses = survey_responses.drop(columns=['MainBranch']) #Removes column 'MainBranch'

In [229]:
## Deal w/age object column which has ranges vs actual ages.  convert to integer ages in new 'age_numeric' column
## To avoid too many columns, will use median age points to create integer based row

# Create age mapping dictionary (using midpoint of each range)
age_mapping = {
    'Under 18 years old': 17,
    '18-24 years old': 21,  # (18+24)/2
    '25-34 years old': 29.5,
    '35-44 years old': 39.5,
    '45-54 years old': 49.5,
    '55-64 years old': 59.5,
    '65 years or older': 70,
    'Prefer not to say': None  
}

survey_responses['age_numeric'] = survey_responses['Age'].map(age_mapping)

In [231]:
# Deal w/No Responses or NaN data points in newly created 'age_numeric' column
median_age = survey_responses['age_numeric'].median()
survey_responses['age_numeric'] = survey_responses['age_numeric'].fillna(median_age)

In [233]:
survey_responses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Columns: 101 entries, ResponseId to age_numeric
dtypes: bool(13), float64(12), int64(1), object(75)
memory usage: 44.7+ MB


In [235]:
survey_responses = survey_responses.drop(columns=['Age'])

In [237]:
### Deal w/responses from 'Employment' column
### Will create boolean columns consistent with other responses
## Strip leading and trailing spaces from responses

survey_responses['Employment'] = survey_responses['Employment'].str.strip()

In [239]:
## create employment mapping dictionary to reference in loopin column creation

employment_mapping = {
    'Employed, full-time': 'employed_full_time',
    'Employed, part-time': 'employed_part_time',
    'Independent contractor, freelancer, or self-employed': 'self_employed',
    'Not employed, but looking for work': 'unemployed_seeking',
    'Not employed, and not looking for work': 'unemployed_not_seeking',
    'Student, full-time': 'student_full_time',
    'Student, part-time': 'student_part_time',
    'Retired': 'retired',
    'I prefer not to say': 'prefer_not_to_say'
}

# Create columns boolean in nature
for status, col_name in employment_mapping.items():
    survey_responses[f'{col_name}'] = (survey_responses['Employment'] == status)

In [241]:
survey_responses.info(100)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 109 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    MainBranch                                                                                   object 
 2    Employment                                                                                   object 
 3    RemoteWork                                                                                   object 
 4    CodingActivities                                                                             object 
 5    EdLevel                                                                                      object 
 6    LearnCode                   

In [243]:
survey_responses = survey_responses.drop(columns=['is_i_am_a_developer_by_profession'])
survey_responses = survey_responses.drop(columns=['is_i_am_not_primarily_a_developer_but_i_write_code_sometimes_as_part_of_my_work_studies'])
survey_responses = survey_responses.drop(columns=['is_i_used_to_be_a_developer_by_profession_but_no_longer_am'])
survey_responses = survey_responses.drop(columns=['is_i_am_learning_to_code'])
survey_responses = survey_responses.drop(columns=['is_i_code_primarily_as_a_hobby'])
survey_responses = survey_responses.drop(columns=['is_no_experience'])


In [245]:
survey_responses.info(100)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 103 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    MainBranch                                                                                   object 
 2    Employment                                                                                   object 
 3    RemoteWork                                                                                   object 
 4    CodingActivities                                                                             object 
 5    EdLevel                                                                                      object 
 6    LearnCode                   

In [247]:
### Remove column 'Employment'
survey_responses = survey_responses.drop(columns=['Employment'])

In [249]:
survey_responses.info(10)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 102 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    MainBranch                                                                                   object 
 2    RemoteWork                                                                                   object 
 3    CodingActivities                                                                             object 
 4    EdLevel                                                                                      object 
 5    LearnCode                                                                                    object 
 6    LearnCodeOnline             

In [251]:
## Deal with 'RemoteWork' column
## Address NA's in column
## Convert 'NAs' to "Remote Not Specified

survey_responses['RemoteWork'] = survey_responses['RemoteWork'].fillna('Remote Not specified')

In [253]:
### Create new boolean columns for each of the response options under'RemoteWork'

remote_responses = [
    'Remote',
    'In-person',
    'Hybrid (some remote, some in-person)',
    'Remote Not Specified']

for responses in remote_responses:
    # Clean mode name for column naming
    col_name = responses.lower()\
        .replace(' ', '_')\
        .replace('(', '')\
        .replace(')', '')\
        .replace('-', '')
    
    # Create boolean column
    survey_responses[f'{col_name}'] = (survey_responses['RemoteWork'] == responses)



In [255]:
survey_responses.info(150)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 106 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    MainBranch                                                                                   object 
 2    RemoteWork                                                                                   object 
 3    CodingActivities                                                                             object 
 4    EdLevel                                                                                      object 
 5    LearnCode                                                                                    object 
 6    LearnCodeOnline             

In [261]:
survey_responses = survey_responses.drop(columns=['MainBranch', 'RemoteWork'])

Deal With 'CodingActivities' column

In [263]:
## create employment mapping dictionary to reference in loopin column creation

afterwork_mapping = {
    'Hobby': 'hobby',
    'Freelance/contract work': 'freelance_contract',
    'Independent contractor, freelancer, or self-employed': 'self_employed',
    'Contribute to open-source projects': 'open_source_contributor',
    'Bootstrapping a business': 'business_bootstrapper',
    'School or academic work': 'school',
    'Professional development or self-paced learning from online courses': 'online_professional_development',
    'I don’t code outside of work': 'no_coding_after_work'
}

# Create columns boolean in nature
for status, col_name in afterwork_mapping.items():
    survey_responses[f'{col_name}'] = (survey_responses['CodingActivities'] == status)

In [265]:
survey_responses.info(120)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 111 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    CodingActivities                                                                             object 
 2    EdLevel                                                                                      object 
 3    LearnCode                                                                                    object 
 4    LearnCodeOnline                                                                              object 
 5    TechDoc                                                                                      object 
 6    YearsCode                   

In [267]:
survey_responses=survey_responses.drop(columns=['CodingActivities'])

In [271]:
survey_responses.info(100)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 110 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    EdLevel                                                                                      object 
 2    LearnCode                                                                                    object 
 3    LearnCodeOnline                                                                              object 
 4    TechDoc                                                                                      object 
 5    YearsCode                                                                                    object 
 6    YearsCodePro                

Deal with EdLevel Column

In [273]:
# Define education level options
ed_levels = [
    'Primary/elementary school',
    'Secondary school (e.g. American high school, German Realschule or Gymnasium, etc.)',
    'Some college/university study without earning a degree',
    'Associate degree (A.A., A.S., etc.)',
    'Bachelor’s degree (B.A., B.S., B.Eng., etc.)',
    'Master’s degree (M.A., M.S., M.Eng., MBA, etc.)',
    'Professional degree (JD, MD, Ph.D, Ed.D, etc.)',
    'Something else'
]

# Create boolean columns
for level in ed_levels:
    # Generate clean column name
    col_name = (
        level.lower()
        .replace(' ', '_')
        .replace('/', '_')
        .replace('(', '')
        .replace(')', '')
        .replace('.', '')
        .replace(',', '')
        .replace('’', '')
        .replace('etc', '')
        .replace('e.g.', '')
        .strip('_')
    )
    
    # Create boolean column
    survey_responses[f'ed_{col_name}'] = survey_responses['EdLevel'] == level

# Deal with missing values or "NA's"
survey_responses['ed_missing'] = survey_responses['EdLevel'].isna()



In [277]:
survey_responses.info(100)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 119 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    EdLevel                                                                                      object 
 2    LearnCode                                                                                    object 
 3    LearnCodeOnline                                                                              object 
 4    TechDoc                                                                                      object 
 5    YearsCode                                                                                    object 
 6    YearsCodePro                

In [285]:
survey_responses = survey_responses.drop(columns=['EdLevel'])

KeyError: "['EdLevel'] not found in axis"

In [289]:
survey_responses.info(120)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 118 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    LearnCode                                                                                    object 
 2    LearnCodeOnline                                                                              object 
 3    TechDoc                                                                                      object 
 4    YearsCode                                                                                    object 
 5    YearsCodePro                                                                                 object 
 6    DevType                     

Deal with column 'LearnCode'

In [291]:


# Define learning methods (from your list)
learn_methods = [
    'Books / Physical media',
    'Coding Bootcamp',
    'Colleague',
    'Friend or family member',
    'Online Courses or Certification',
    'On the job training',
    'Other online resources (e.g., videos, blogs, forum, online community)',
    'School (i.e., University, College, etc)',
    'Other (please specify)'
]

# Clean method names for column creation
def clean_method_name(method):
    return (
        method.lower()
        .replace(' ', '_')
        .replace('/', '_')
        .replace('(', '')
        .replace(')', '')
        .replace('.', '')
        .replace(',', '')
        .replace('e.g.', '')
        .replace('i.e.', '')
        .strip('_')
    )

# Standardize delimiters and split responses
def split_learn_code(responses):
    if pd.isna(responses):
        return []
    # Replace semicolons with commas, then split
    return [x.strip() for x in responses.replace(';', ',').split(',')]

# Create boolean columns
for method in learn_methods:
    col_name = f"learn_{clean_method_name(method)}"
    survey_responses[col_name] = (
        survey_responses['LearnCode']
        .apply(lambda x: method in split_learn_code(x))
    )

# Handle NA values
survey_responses['learn_missing'] = survey_responses['LearnCode'].isna()





In [295]:
survey_responses.head(20)

ResponseId                                          LearnCode  \
0            1                             Books / Physical media   
1            2  Books / Physical media;Colleague;On the job tr...   
2            3  Books / Physical media;Colleague;On the job tr...   
3            4  Other online resources (e.g., videos, blogs, f...   
4            5  Other online resources (e.g., videos, blogs, f...   
5            6  School (i.e., University, College, etc);Online...   
6            7  Other online resources (e.g., videos, blogs, f...   
7            8  Other online resources (e.g., videos, blogs, f...   
8            9  Books / Physical media;Other online resources ...   
9           10  On the job training;Other online resources (e....   
10          11  Books / Physical media;Other online resources ...   
11          12  Books / Physical media;School (i.e., Universit...   
12          13  Books / Physical media;On the job training;Sch...   
13          14  Colleague;Other online resources (e.g., videos...   
14          15  Books / Physical media;Other online resources ...   
15          16  Books / Physical media;On the job training;Oth...   
16          17  Books / Physical media;Other online resources ...   
17          18  Other online resources (e.g., videos, blogs, f...   
18          19  Books / Physical media;On the job training;Oth...   
19          20  Books / Physical media;Other online resources ...   

                                      LearnCodeOnline  \
0                                                 NaN   
1   Technical documentation;Blogs;Books;Written Tu...   
2   Technical documentation;Blogs;Books;Written Tu...   
3   Stack Overflow;How-to videos;Interactive tutorial   
4   Technical documentation;Blogs;Written Tutorial...   
5                                                 NaN   
6   Technical documentation;Stack Overflow;Written...   
7   Technical documentation;Video-based Online Cou...   
8         Stack Overflow;Written-based Online Courses   
9   Technical documentation;Blogs;Written Tutorial...   
10  Technical documentation;Books;Written Tutorial...   
11                                                NaN   
12                                                NaN   
13  Technical documentation;Blogs;Stack Overflow;I...   
14  Stack Overflow;Coding sessions (live or record...   
15  Technical documentation;Books;Written Tutorial...   
16  Technical documentation;Written Tutorials;Stac...   
17  Blogs;Coding sessions (live or recorded);How-t...   
18  Technical documentation;Blogs;Books;Stack Over...   
19  Technical documentation;Blogs;Books;Written Tu...   

                                              TechDoc YearsCode YearsCodePro  \
0                                                 NaN       NaN          NaN   
1   API document(s) and/or SDK document(s);User gu...        20           17   
2   API document(s) and/or SDK document(s);User gu...        37           27   
3                                                 NaN         4          NaN   
4   API document(s) and/or SDK document(s);User gu...         9          NaN   
5                                                 NaN        10          NaN   
6                                                 NaN         7            7   
7                          First-party knowledge base         1          NaN   
8                                                 NaN        20          NaN   
9   Traditional public search engine;AI-powered se...        15           11   
10  API document(s) and/or SDK document(s);User gu...        20          NaN   
11                                                NaN        30           25   
12                                                NaN        20           12   
13  API document(s) and/or SDK document(s);User gu...        20          NaN   
14                                                NaN        15           10   
15             API document(s) and/or SDK document(s)        31           27

In [297]:
survey_responses = survey_responses.drop(columns=['LearnCode'])

In [299]:
survey_responses.info(120)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65437 entries, 0 to 65436
Data columns (total 127 columns):
 #    Column                                                                                       Dtype  
---   ------                                                                                       -----  
 0    ResponseId                                                                                   int64  
 1    LearnCodeOnline                                                                              object 
 2    TechDoc                                                                                      object 
 3    YearsCode                                                                                    object 
 4    YearsCodePro                                                                                 object 
 5    DevType                                                                                      object 
 6    Country                     